yksgi/src/lib.rs

```diff
@@ -149,6 +168,271 @@ impl SGInterp {
         }
     }
 
+    // Generate frames from stackmap information after a guard failure. The new frames are stored
+    // inside some allocated memory whose pointer this function returns. The frames are then later
+    // `memcpy`ed to the actual stack by ykcapi::__ykrt_reconstruct_frames.
+    pub fn reconstruct_frames(&self, btmframeaddr: *mut c_void) -> *const c_void {
+        // Load the stackmap and text sections from the binary. The former we need to parse in the
+        // stackmaps. The latter is required to disassemble the binary in order to distinguish
+        // branches from calls.
+        // FIXME: Don't use current_exe.
```

Comment:
OK, this is where object is used. We shouldn't need to look into the binary file for either `.text` or the stackmap section. They are both loaded now. This could be a separate PR if @ltratt is OK with that.

We might also be able to avoid instantiating capstone using some magic. What do you need? the size of the instruction in the AOT code which could become a guard failure in the JIT code?

Reply:
Yes I probably need @vext01 help to get rid of the `current_exe`. Will do a follow up PR immediately once this is in.

Regarding the capstone thing. I need to size of the instruction before stackmap. Which may be a `call` or an `icmp` or `test`. If it's a call, I need its size in order to adjust the record offset.

Reply:
We may be able to insert something at compile-time, but let's roll with this for now.

Perhaps add a FIXME?

Reply:
I'd say roll with capstone and deal with a better way in a follow up PR.

Reply:
I seem to be saying this a lot, but I agree with Edd.

Reply:
I concur. This will likely require a ykllvm change too. So better for another time.

Reply:
Added a fixme: [9294cdb](https://github.com/ykjit/yk/pull/578/commits/9294cdb1fea1440e857e172f981b9a5886501ac2)